developer/src/kmc/src/commands/buildClasses/BuildKeyboardInfo.ts

```diff
@@ -19,15 +25,76 @@ export class BuildKeyboardInfo extends BuildActivity {
       return false;
     }
 
-    // let outputFilename: string = this.getOutputFilename(infile, options);
-    // let code = null;
+    const metadata = findProjectFile(callbacks, project, KeymanFileTypes.Source.KeyboardInfo);
+    if(!metadata) {
+      // Project loader should always have added a metadata file
+      return false;
+    }
+
+    if(!fs.existsSync(project.resolveInputFilePath(metadata))) {
+      // For now, if the metadata file does not exist, we won't attempt to build
+      // it. One day in the future, when source metadata files become optional,
+      // we'll need to skip this
+      return true;
+    }
+
+
+    const keyboard = findProjectFile(callbacks, project, KeymanFileTypes.Source.KeymanKeyboard);
+    const kps = findProjectFile(callbacks, project, KeymanFileTypes.Source.Package);
+    if(!keyboard || !kps)  {
+      return false;
+    }
+
+    let kmpCompiler = new KmpCompiler(callbacks);
+    let kmpJsonData = kmpCompiler.transformKpsToKmpObject(project.resolveInputFilePath(kps));
```

Comment:
Wait... so it's an object we're only building within the calling method?  Why _isn't_ `kmpJsonData` a direct parameter of `writeMergedKeyboardInfoFile`, then?  I don't recall seeing it as an, uh, "optional option" within that method.

Reply:
Will do.